dev-docs/design/ssr-react.md

```diff
@@ -0,0 +1,72 @@
+# Server Rendering Lit Components in React
+
+## Background
+
+React server rendering of custom elements will only shallow render components by emitting the host element tag with props stringified to attributes and any light DOM children it has. `@lit-labs/ssr` is able to server render Lit elements deeply by emitting the contents of the component as Declarative Shadow DOM. We want usage of custom elements in React with server rendering to be able to take advantage of this as well.
+
+Client rendering of custom elements in React have their shortfalls such as all props being set as attributes on the element and adding event listeners being clunky. We have the `createComponent` in `@lit-labs/react` that creates a React Component that wraps the custom element which helps with this situation.
+
+## Scenarios to Handle
+
+<!-- prettier-ignore-start -->
+| | With vanilla React | What we want |
+| --- | --- | --- |
+| Bare custom elements | <p>Here we simply rely on React's built in SSR which will render the custom element tag, setting all props as attributes. There's no content until JS with component definition is loaded in the client and the custom element is upgraded.</p><p>There is potential for FOUC when the component has light DOM children as these won't have a slot to project to and show up unstyled in the initial document before upgrade, and content shifts as upgraded components fill up spaces.</p> | <p>To mirror the client-side React behavior, all props will be set as attributes on the element as it is server rendered with `@lit-labs/ssr`.<p><p>This will work fine for simple elements whose properties are all serializable and the attribute name matches the property name. More complex components may fail to render correctly.</p> |
```

Comment:
Nit: Add (Flash of unstyled content) in brackets or something to the first time the FOUC acronym is used. May not be an immediately familiar acronym to the reader.